src/relay/op/tensor/transform.cc

```diff
@@ -454,6 +454,61 @@ RELAY_REGISTER_OP("full")
 .set_support_level(3)
 .add_type_rel("Full", FullRel);
 
+bool TensorRel(const Array<Type>& types,
+               int num_inputs,
+               const Attrs& attrs,
+               const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 1);
+  const TensorAttrs* param = attrs.as<TensorAttrs>();
+
+  reporter->Assign(types[0], TensorTypeNode::make(param->shape, param->dtype));
+  return true;
+}
+
+Expr MakeZeros(Array<IndexExpr> shape,
+               DataType dtype) {
+  auto attrs = make_node<TensorAttrs>();
+  attrs->shape = std::move(shape);
+  attrs->dtype = std::move(dtype);
+  static const Op& op = Op::Get("zeros");
+  return CallNode::make(op, {}, Attrs(attrs), {});
+}
+
+TVM_REGISTER_API("relay.op._make.zeros")
+.set_body([](const TVMArgs& args, TVMRetValue* rv) {
+    runtime::detail::unpack_call<Expr, 2>(MakeZeros, args, rv);
+  });
+
+RELAY_REGISTER_OP("zeros")
+.describe(R"code(Fill array with zeros.
+
+)code" TVM_ADD_FILELINE)
+.set_num_inputs(0)
+.set_support_level(3)
+.add_type_rel("Tensor", TensorRel);
+
+Expr MakeOnes(Array<IndexExpr> shape,
+              DataType dtype) {
+  auto attrs = make_node<TensorAttrs>();
```

Comment:
Nit: I know this is also used in the other places of the code base, but I would recommend we use `auto*` when it is a pointer to reduce confusion.

Reply:
it is impossible as make_node return a nodeptr, instead of a raw pointer.

Reply:
hmm, I think it is should be possible to return a nodeptr, e.g. `NodePtr<TensorAttrs> attrs = make_node<TensorAttrs>()` should work, but I agree it is more verbose.